drivers/timer/xlnx_psttc_timer.c

```diff
@@ -1,187 +1,186 @@
 /*
+ * Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
  * Copyright (c) 2018 Xilinx, Inc.
+ *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <kernel.h>
+#include <soc.h>
 #include <drivers/timer/system_timer.h>
+#include "xlnx_psttc_timer_priv.h"
 
-#ifdef CONFIG_TICKLESS_KERNEL
-#error "Tickless mode is not supported"
-#endif
+#define TIMER_INDEX		CONFIG_XLNX_PSTTC_TIMER_INDEX
+#define TIMER_DT(v)		UTIL_CAT(UTIL_CAT(DT_INST_, TIMER_INDEX), _##v)
 
-#define TIMER_FREQ          CONFIG_SYS_CLOCK_TICKS_PER_SEC
+#define TIMER_IRQ		TIMER_DT(XLNX_TTCPS_IRQ_0)
+#define TIMER_BASE_ADDR		TIMER_DT(XLNX_TTCPS_BASE_ADDRESS)
+#define TIMER_CLOCK_FREQUECY	TIMER_DT(XLNX_TTCPS_CLOCK_FREQUENCY)
 
-#if (CONFIG_XLNX_PSTTC_TIMER_INDEX == 0)
-#define TIMER_INPUT_CLKHZ   DT_INST_0_CDNS_TTC_CLOCK_FREQUENCY
-#define TIMER_IRQ           DT_INST_0_CDNS_TTC_IRQ_0
-#define TIMER_BASEADDR      DT_INST_0_CDNS_TTC_BASE_ADDRESS
-#else
-#error ("No timer is specified")
-#endif
+#define TICKS_PER_SEC		CONFIG_SYS_CLOCK_TICKS_PER_SEC
+#define CYCLES_PER_SEC		TIMER_CLOCK_FREQUECY
+#define CYCLES_PER_TICK		(CYCLES_PER_SEC / TICKS_PER_SEC)
+#define CYCLES_NEXT_MIN		(10000)
+#define CYCLES_NEXT_MAX		(4294967295)
```

Comment:
UINT_MAX would be clearer

Reply:
sure, I just realised there is also `XTTC_MAX_INTERVAL_COUNT` in `xlnx_psttc_timer_priv.h`.

I will replace the constant with that symbol since that is part of the "peripheral definition" header.